common/vm_take_screenshot.yml

```diff
@@ -2,22 +2,81 @@
 # SPDX-License-Identifier: BSD-2-Clause
 ---
 # Capture the specified VM console screenshot and transfer screenshot
-# file to local path if 'vm_take_screenshot_local_path' is defined.
+# file to local path if 'vm_screenshot_local_dir' is defined.
 # Parameters:
-#   vm_take_screenshot_path(optional): the local path of the screenshot file
+#   vm_screenshot_local_dir(optional): the local dir path for the screenshot taken from VM
+#   vm_screenshot_local_name(optional): the local file name for the VM screenshot
+#   vm_screen_active(optional): whether to send a key to active VM screen. Default is false.
+# Return:
+#   vm_screenshot_local_path: the local file path of the screenshot taken from VM
 #
-- name: "Take VM '{{ vm_name }}' console screenshot"
-  community.vmware.vmware_guest_screenshot:
-    validate_certs: "{{ validate_certs | default(false) }}"
-    hostname: "{{ vsphere_host_name }}"
-    username: "{{ vsphere_host_user }}"
-    password: "{{ vsphere_host_user_password }}"
-    datacenter: "{{ vsphere_host_datacenter }}"
-    folder: "{{ vm_folder }}"
-    name: "{{ vm_name }}"
-    local_path: "{{ vm_take_screenshot_local_path | default(omit) }}"
-  register: vm_take_screnshot_result
-  failed_when: false
-- name: Display taking VM screenshot result
-  ansible.builtin.debug: var=vm_take_screnshot_result
-  when: enable_debug is defined and enable_debug
+- name: "Initialize the fact of local file path for the VM screenshot"
+  ansible.builtin.set_fact:
+    vm_screenshot_local_path: ""
+
+- name: "Get VM power state"
+  include_tasks: vm_get_power_state.yml
+
+- name: "VM is not 'poweredOn', couldn't take screenshot"
```

Comment:
It's because the Ansible module community.vmware.vmware_guest_screenshot can only work with PoweredOn VM, or just to take the screenshot for PoweredOn VM?

Reply:
Ummm....Think about such kind of scenario. Power off you laptop, and then you take a screenshot on it. Is it possible?

Reply:
When the util is called by test_rescue, it has meaning to take snaphsot for powered off VM. It catches the state by that time. For this scenario, if powering on VM, will destroy the clue and lost the correct state when issue occurs.

Reply:
There is no power-on action.